### DISCUSSION TITLE
Use an Ansible group for TDP cluster instead of "all"

### DISCUSSION THREAD

**Discussion by rpignolet** [TOSIT-IO/TDP#49] [Polls]
In Kubespray inventory sample (https://github.com/kubernetes-sigs/kubespray/blob/master/inventory/sample/inventory.ini) they use a group `k8s_cluster` used by Kubespray playbooks to know what Ansible hosts are part of Kubernetes cluster.

Inside our actual getting started, we use `all` Ansible groups to configure `/etc/hosts` and `yum` packages for example. I think we should create an Ansible group `tdp_cluster` or `hadoop_cluster` and never use the `all` Ansible groups to not configure hosts which are not part of TDP/Hadoop cluster.

**Comment by gboutry** [comment]
I'm all for having a separate entity designating the cluster, moreover, I would propose we move the cluster definitions vars that were inside the group `hadoop` towards a group with the new name.

The plan, for now, was to move them to the `all` group.

**Comment by leopaul36** [comment]
Agreed, it would make sense for this file https://github.com/TOSIT-IO/tdp-getting-started/blob/master/inventory/group_vars/all.yml to become `tdp_cluster.yml`

**Comment by jusstol** [comment]
I completely agree. On our side at RTE, we created the `groupnodes` group to our inventory just to be able to put every nodes as children of a specific environment (all nodes are children of C03, which is child of OPF). We had to create this group because children of groups must be groups, and using the special group `all` is not allowed because all groups are already children of `all` :
```
ERROR! Adding group 'OPF' as child to 'all' creates a recursive dependency loop.
```
Our inventory :
```ini
....

[groupnodes]
worker-01
worker-02
worker-03
master-01
master-02
master-03
edge-01

[C03:children]
groupnodes

[OPF:children]
C03
```

**Comment by rpignolet** [comment]
Poll result -> use `tdp_cluster` instead of `all`.
